File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/utils/Dialogs.java
package com.sorcerer.sorcery.iconpack.ui.utils;

import android.content.Context;

import com.afollestad.materialdialogs.MaterialDialog;
import com.sorcerer.sorcery.iconpack.R;
import com.sorcerer.sorcery.iconpack.utils.ResourceUtil;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2017/2/17
 */

public class Dialogs {
    public static MaterialDialog.Builder builder(Context context) {
        return new MaterialDialog.Builder(context)
                .backgroundColor(ResourceUtil.getAttrColor(context, R.attr.colorCard));
    }

    public static MaterialDialog indeterminateProgressDialog(Context context,
                                                             String message) {
        return new MaterialDialog.Builder(context)
                .progress(true, 0)
                .content(message)
                .canceledOnTouchOutside(false)
                .build();
    }

    public static MaterialDialog horizontalIndeterminateProgressDialog(Context context,
                                                                       String message) {
        return new MaterialDialog.Builder(context)
                .progress(true, 0)
                .content(message)
                .canceledOnTouchOutside(false)
                .progressIndeterminateStyle(true)
                .build();
    }
}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/utils/Dialogs.kt
package com.sorcerer.sorcery.iconpack.ui.utils

import android.content.Context

import com.afollestad.materialdialogs.MaterialDialog
import com.sorcerer.sorcery.iconpack.R
import com.sorcerer.sorcery.iconpack.utils.ResourceUtil

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2017/2/17
 */

object Dialogs {
    fun builder(context: Context): MaterialDialog.Builder {
        return MaterialDialog.Builder(context)
                .backgroundColor(ResourceUtil.getAttrColor(context, R.attr.colorCard))
    }

    fun indeterminateProgressDialog(context: Context,
                                    message: String): MaterialDialog {
        return MaterialDialog.Builder(context)
                .progress(true, 0)
                .content(message)
                .canceledOnTouchOutside(false)
                .build()
    }

    fun horizontalIndeterminateProgressDialog(context: Context,
                                              message: String): MaterialDialog {
        return MaterialDialog.Builder(context)
                .progress(true, 0)
                .content(message)
                .canceledOnTouchOutside(false)
                .progressIndeterminateStyle(true)
                .build()
    }
}
